common/app/common/dfp/TagSponsorship.scala

```diff
@@ -65,6 +65,26 @@ object InlineMerchandisingTargetedTagsReportParser extends Logging {
   }
 }
 
+object HighMerchandisingTargetedTagSet {
+  implicit val jsonReads = Json.reads[HighMerchandisingTargetedTagSet]
+
+  implicit val highMerchandisingTargetedTagSetWrites = new Writes[HighMerchandisingTargetedTagSet] {
+    def writes(tagSet: HighMerchandisingTargetedTagSet): JsValue = {
+      Json.obj(
+        "items" -> tagSet.items
+      )
+    }
+  }
+}
+
+case class HighMerchandisingTargetedTagSet(items: Set[String] = Set.empty){
+  
+  def hasTag (tag: Tag): Boolean = items.contains(tag.id.split('/').lastOption.get)
+
+  def nonEmpty = items.nonEmpty
+
+}
+
```

Comment:
@rich-nguyen refactored to one def, not using exists function however, Is there a better way using exists?